.github/workflows/build_and_test.yml

```diff
@@ -649,11 +649,11 @@ jobs:
         if [ -f ./dev/free_disk_space_container ]; then
           ./dev/free_disk_space_container
         fi
-    - name: Install Java 8
+    - name: Install Java 17
```

Comment:
I need to think about this part a bit more. I just recalled that branch-3.x will also use this yml, so we may need to add some if conditions.